serial_communication/web/app.py

```diff
@@ -31,43 +36,47 @@ def get_serial_ports():
             for port in serial.tools.list_ports.comports()
             if port.device.startswith('/dev/tty')]
 
-def update_serial_port():
-    """Update the serial port dynamically."""
-    try:
-        global BG_TASK, ser # pylint: disable=global-statement
-        max_port_number = 5  # Maximum port number to try
-        port_pattern = '/dev/ttyACM{}'
-        print("Trying to connect to port!")
-        while True:
-            for port_number in range(max_port_number):
-                port = port_pattern.format(port_number)
-                try:
-                    temp_ser = serial.Serial(port, baudrate=115200, timeout=1)
-                    ser = temp_ser
-                    print(f"Connected to {port}")
-                    set_serial_object(ser)
-                    BG_TASK = True
-                    return temp_ser
-                except serial.SerialException:
-                    print(f"Port {port} not available. Trying the next one.")
-
-            print(f"No available ports (tried up to {max_port_number}). Retrying in 10 seconds...")
-            time.sleep(10)
-            update_serial_port()
-    except Exception as usp_e: # pylint: disable=broad-except
-        exception_logger("update_serial_port", usp_e)
-        return None
-
 # Replace with the default serial port
-ser = update_serial_port()
+ser = update_serial_port(TTGO_TCALL_PORT)
+set_serial_object(ser)
+if COMMUNICATION_PORT is None:
+    COMMUNICATION_PORT = ser
+# ser = serial.Serial('/dev/ttyUSB0', 115200)
 
 @app.route('/')
 def index():
     """Render the index page."""
     available_ports = get_serial_ports()
-    baud_rates = [9600, 115200, 230400, 460800, 921600]
+
+    if COMMUNICATION_PORT:
+        print(f"\n\n\t\t sending to default index.html: {COMMUNICATION_PORT}")
+        return render_template('index.html',
+                               default_port=COMMUNICATION_PORT,
+                               available_ports=available_ports)
+    print("\n\n\t\t !!!___Sending default to index.html___!!!")
     return render_template('index.html',
-                default_port=ser, available_ports=available_ports, baud_rates=baud_rates)
+                            default_port=ser,
+                            available_ports=available_ports)
+
+@app.route('/update_port', methods=['POST'])
+def update_port():
+    """To update port dynamically from the web page"""
+    try:
+        selected_port = request.get_json().get('port')
+        if selected_port is not None:
+            print("Trying to change to: " + selected_port)
+            global ser, COMMUNICATION_PORT # pylint: disable=global-statement
+            ser = serial.Serial(selected_port, 115200)
+            COMMUNICATION_PORT = ser
```

Comment:
@mhamidjamil we got problem here, if you change this port then you communication will be disturbed with ttgo which we don't want to, so please update the port in such a way so the communication with the ttgo module should also not disturb.

Reply:
You are write like if we forget to set back the TTGO port then the communication will be unexpected.

Reply:
I have verified serial_handler is just responsible for communication with ttgo so no need to worry about ttgo communication i also have verified it from my side by changing port from web and then leave the page after rebooting ttgo the prayer time was updated after 5 minutes.